### DISCUSSION TITLE
ArgSegment not updating

### DISCUSSION THREAD

**Discussion by tbibre** [AvaloniaUI/Avalonia#12037] [Q&A]
Hi,

I'm trying to change the length of an ArcSegment (and later, I'll try to use Transitions between its states), but it isn't redrawn when I change the end point.

Here is the xaml:
```
<Path Stroke="Red"
	  StrokeThickness="24">
	<Path.Data>
		<PathGeometry>
			<PathGeometry.Figures>
				<PathFigure StartPoint="520,230"
							IsClosed="False">
					<PathFigure.Segments>
						<ArcSegment Size="290,290"
									IsLargeArc="True"
									SweepDirection="CounterClockwise"
									Point="{CompiledBinding CircleEndPoint}"/>
					</PathFigure.Segments>
				</PathFigure>
			</PathGeometry.Figures>
		</PathGeometry>
	</Path.Data>
</Path>
```
And C# (with some harcoded things as I'm still testing)
```
public Avalonia.Point CircleEndPoint
{
	get
	{
		Avalonia.Point Origin = new Avalonia.Point(520, 520);
		Avalonia.Point Start = new Avalonia.Point(Origin.X, Origin.Y -290);

		double Angle = (360 - (360 * Speed) / 300.0) * (Math.PI / 180.0);

		double XPos = (Start.X - Origin.X) * Math.Cos(-Angle) - (Start.Y - Origin.Y) * Math.Sin(-Angle);
		double YPos = (Start.Y - Origin.Y) * Math.Cos(-Angle) + (Start.X - Origin.X) * Math.Sin(-Angle);

		return new Avalonia.Point(XPos + Origin.X, YPos + Origin.Y);
	}
}
```

The property changed is send when the Speed changes. If I put a breakpoint there, it is correctly hit when the value change.
The code seems ok, as the Arc has the right shape from whatever the initialization value is.

I've seen some previous issues about this, but it seems to be fixed a long time ago.
https://github.com/AvaloniaUI/Avalonia/issues/3952

**Comment by timunie** [comment]
Which Avaonia version? It's always important to let us know. Iirc should work in 11.0

**Reply by tbibre** [reply]
Oups sorry, it was 11-rc2.2. I'll try with 11.0 today.

**Reply by timunie** [reply]
nah then it will not be updated.

**Reply by timunie** [reply]
Okay, looks like this is an actual bug. I'll create an issue for it to track. Thanks for the sample 👍
